ext/sockets/sockets.c

```diff
@@ -1883,11 +1887,21 @@ PHP_FUNCTION(socket_set_option)
 				RETURN_THROWS();
 			}
 
-			convert_to_long(l_onoff);
-			convert_to_long(l_linger);
+			zend_long vall_lonoff = zval_get_long(l_onoff);
+			zend_long vall_linger = zval_get_long(l_linger);
```

Comment:
Not sure why it is `vall`?
```suggestion
			zend_long val_lonoff = zval_get_long(l_onoff);
			zend_long val_linger = zval_get_long(l_linger);
```